Source/DafnyCore/Resolver/NameResolutionAndTypeInference.cs

```diff
@@ -5989,7 +5993,11 @@ public class ResolveTypeReturn {
       } else {
         // ----- None of the above
         if (complain) {
-          reporter.Error(MessageSource.Resolver, expr.tok, "unresolved identifier: {0}", name);
+          if (resolutionContext.InReveal) {
+            reporter.Error(MessageSource.Resolver, expr.tok, "cannot reveal '{0}' because no constant, assert label or requires label in the current scope is named '{0}'", expr.Name);
```

Comment:
```suggestion
            reporter.Error(MessageSource.Resolver, expr.tok, "cannot reveal '{0}' because no constant, assert label, or requires label in the current scope is named '{0}'", expr.Name);
```